src/test/java/com/puppycrawl/tools/checkstyle/DetailAstImplTest.java

```diff
@@ -493,7 +492,9 @@ public void testAddNextSiblingNullParent() {
         assertWithMessage("Invalid next sibling")
             .that(newSibling.getNextSibling())
             .isNull();
-        assertSame(newSibling, child.getNextSibling(), "Invalid child");
+        assertWithMessage("Invalid parent")
+            .that(child.getNextSibling())
+            .isEqualTo(newSibling);
```

Comment:
This should be same instance

Reply:
Why is that? DetailAST is singleton and equals is the same as saying its the same instance.

Reply:
In this case we do not want check by equals method, we want to make sure objects are same by reference id .

Reply:
The equals method does this currently since we don't override it.

This was changed.